pulsar/producer_partition.go

```diff
@@ -464,77 +469,82 @@ func (p *partitionProducer) reconnectToBroker(connectionClosed *connectionClosed
 	)
 
 	for maxRetry != 0 {
-		if p.getProducerState() != producerReady {
-			// Producer is already closing
-			p.log.Info("producer state not ready, exit reconnect")
+		select {
+		case <-p.ctx.Done():
```

Comment:
the default case is too long, may be below is better
select {
case <-p.ctx.Done()
    return
default:
}

Reply:
Done.